packages/bsky/src/views/index.ts

```diff
@@ -1111,6 +1131,462 @@ export class Views {
+    }
+
+    const childrenUris = childrenByParentUri[parentUri] ?? []
+    return mapDefined(childrenUris, (uri) => {
+      // Not found.
+      const post = state.posts?.get(uri)
+      if (post?.violatesThreadGate) {
+        return undefined
+      }
+      const postView = this.post(uri, state)
+      if (!post || !postView) {
+        return undefined
+      }
+      const authorDid = postView.author.did
+      if (rootUri !== getRootUri(uri, post)) {
+        // outside thread boundary
+        return undefined
+      }
+
+      // Blocked (1p and 3p for replies).
+      const has1pBlock = this.viewerBlockExists(authorDid, state)
+      const has3pBlock =
+        !state.ctx?.include3pBlocks && state.postBlocks?.get(uri)?.parent
+      if (has1pBlock || has3pBlock) {
+        return undefined
+      }
+      if (!this.viewerSeesNeedsReview({ uri, did: authorDid }, state)) {
+        return undefined
+      }
+
+      // Muted (omits replies except for anchor replies)
+      if (depth > 1 && this.viewerMuteExists(authorDid, state)) {
+        return undefined
+      }
+
+      // Hidden (omits replies except for anchor replies)
+      if (depth > 1 && this.replyIsHiddenByThreadgate(uri, rootUri, state)) {
+        return undefined
+      }
+
+      // No unauthenticated.
+      if (this.noUnauthenticatedPost(viewer, postView)) {
+        return undefined
+      }
+
+      // Recurse down.
+      const isOPThread = isOpThreadFromRootToParent && authorDid === opDid
+      const replies = this.threadV2Replies({
+        parentUri: uri,
+        viewer,
+        isOPThread,
+        opDid,
+        rootUri,
+        childrenByParentUri,
+        state,
+        below,
+        depth: depth + 1,
+      })
+
+      const tree: ThreadTree = {
+        item: this.threadV2ItemPost(
+          {
+            depth,
+            isOPThread,
+            postView,
+            rootUri,
+            uri,
+            viewer,
+          },
+          state,
+        ),
+        hasOPLike: !!state.threadContexts?.get(postView.uri)?.like,
+        parent: undefined,
+        replies,
+      }
+
+      return tree
+    })
+  }
+
+  private threadV2ItemPost(
+    {
+      depth,
+      isOPThread,
+      postView,
+      rootUri,
+      uri,
+      viewer,
+    }: {
+      depth: number
+      isOPThread: boolean
+      postView: PostView
+      rootUri: string
+      uri: string
+      viewer: HydrateCtx['viewer']
+    },
+    state: HydrationState,
+  ): ThreadItemValuePost {
+    const authorDid = creatorFromUri(uri)
+
+    const hasReplies = (postView.replyCount ?? 0) > 0
+    const hiddenByThreadgate =
+      viewer !== authorDid &&
+      this.replyIsHiddenByThreadgate(uri, rootUri, state)
+    const mutedByViewer = this.viewerMuteExists(authorDid, state)
+
+    type Annotation = [boolean, ThreadItemPost['annotations'][number]]
+    const annotations: Annotation[] = [
+      [hasReplies, 'app.bsky.unspecced.getPostThreadV2#hasReplies'],
+      [
```

Comment:
Drive-by thought: it'd be great to have `isAnchor` here too, esp if we ever change depth calculation on the backend.

Reply:
Just curious, did yall ever talk about something like `{ ancestors, anchor, replies }`?

Reply:
I mean't the response structure having three distinct parts (`ancestors`, `anchor`, `replies`), each individual part flat.